### PR TITLE
Add wfc compat for 2 3 x

### PR DESCRIPTION
**Pull request by free-ice**
**Overview**
Add"IssuerSerialQuoteFormat" a variant of ""IssuerSerial" option to output a issuer name in a format that is compatiable with both Microsoft's WCF stack and Java.
This is achieved by using quotes to delimit unicode characters as opposed to a backslash.

This solves the :
**The EncryptedKey clause was not wrapped with the required encryption token 'System.IdentityModel.Tokens.X509SecurityToken'**
error seen in IIS and which dates back to at least 2007.

Example, minimal diff:
```
		ISSUER_SERIAL               OU=Sales,CN=J. Smith
		ISSUER_SERIAL_QUOTE_FORMAT  OU=Sales, CN=J. Smith
```
Example which prompted this change:
```
ISSUER_SERIAL:                 CN=Entrust Certification Authority - L1K,OU=(c) 2012 Entrust\, Inc. - for authorized use only,OU=See www.entrust.net/legal-terms,O=Entrust\, Inc.,C=US

ISSUER_SERIAL_QUOTE_FORMAT:    CN=Entrust Certification Authority - L1K, OU="(c) 2012 Entrust, Inc. - for authorized use only", OU=See www.entrust.net/legal-terms, O="Entrust, Inc.", C=US
```

**Details**
RFC 2253 seems to provide two approaches for escaping characters.
WCF seems to only support the use of using "".
Java seems to support both approaches using \ or using "". 
This claim is supported by Unit tests by testing against all CA in Java's keystore used to test wss4j and also a fixed set of CA.

**Misc**
This PR also consolidates the two different SoapUtils into a single test class.

**Testing**

This  has mainly been tested under Java 8,and against both a test and production WCF system.

### CODE REVIEW

**Comment by free-ice** [comment]
@coheigea  Sorry about noise, rebased change has been tested against two systems , (spring-ws) didn't require and update afterall since string was passed (Flexiable interface design!).

File: ws-security-policy-stax/pom.xml [hunk 1]
```diff
@@ -52,6 +52,13 @@
             <scope>compile</scope>
             <optional>true</optional>
         </dependency>
+        <dependency>
+            <groupId>org.apache.wss4j</groupId>
+            <artifactId>wss4j-ws-security-common</artifactId>
+            <version>${project.version}</version>
+            <classifier>tests</classifier>
+            <scope>test</scope>
+        </dependency>
```
Comment (coheigea) [inline ws-security-policy-stax/pom.xml:61]: This change is not needed.

**Review by coheigea — CHANGES_REQUESTED** [review]
Looks good overall, just some minor changes requested.

File: ws-security-common/src/test/java/org/apache/wss4j/common/util/CommaDelimiterRfc2253NameTest.java [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+	public void testThatACommaDelimitedDnStringAndABackSlashEscapedDnProducesTheSameX509PrincipalUsingDefaultTruststore()
+			throws KeyStoreException, InvalidAlgorithmParameterException, CertificateException, NoSuchAlgorithmException, IOException {
+		KeyStore keystore = loadDefaultKeyStore();
+		assertAllCaTransformsAreEquivalent(keystore);
+	}
+
+	@Test
+	public void testThatACommaDelimitedDnStringAndABackSlashEscapedDnProducesTheSameX509Principal()
+			throws KeyStoreException, InvalidAlgorithmParameterException, CertificateException, NoSuchAlgorithmException, IOException {
+		KeyStore keystore = loadKeyStore("keys/cacerts-openjdk.jks", "changeit");
```
[the remark this answers is not in the document]
Comment (free-ice) [reply ws-security-common/src/test/java/org/apache/wss4j/common/util/CommaDelimiterRfc2253NameTest.java:103]: Done

File: ws-security-dom/src/main/java/org/apache/wss4j/dom/WSConstants.java [hunk 3]
```diff
@@ -319,9 +323,31 @@
      */
     public static final int ENDPOINT_KEY_IDENTIFIER = 14;
 
+
+    /**
+     *Sets the {@link org.apache.wss4j.dom.message.WSSecSignature#build(Crypto)}
+     * or the {@link org.apache.wss4j.dom.message.WSSecEncrypt#build(Crypto, SecretKey)}
+     * method to send the issuer name and the serial number of a certificate to
+     *the receiver.
```
[the remark this answers is not in the document]
Comment (free-ice) [reply ws-security-dom/src/main/java/org/apache/wss4j/dom/WSConstants.java:331]: 👍

File: ws-security-dom/src/main/java/org/apache/wss4j/dom/handler/WSHandlerConstants.java [hunk 4]
```diff
@@ -87,6 +87,8 @@ private WSHandlerConstants() {
      * </li>
      * <li><code>IssuerSerial</code> for {@link WSConstants#ISSUER_SERIAL}
      * </li>
+     * <li><code>IssuerSerial</code> for {@link WSConstants#ISSUER_SERIAL_QUOTE_FORMAT}
```
[the remark this answers is not in the document]
Comment (free-ice) [reply ws-security-dom/src/main/java/org/apache/wss4j/dom/handler/WSHandlerConstants.java:90]: 👍

File: ws-security-dom/src/main/java/org/apache/wss4j/dom/WSConstants.java [hunk 5]
```diff
@@ -191,17 +191,21 @@
...
      * the receiver.
      * <p/>
      * In contrast to {@link #BST_DIRECT_REFERENCE} only the issuer name
      * and the serial number of the signing certificate are sent to the
      * receiver. This reduces the amount of data being sent. The encryption
      * method uses the public key associated with this certificate to encrypt
-     * the symmetric key used to encrypt data.
+     * the symmetric key used to encrypt data. 
+     * The name format will
+     * delimit unicode characters with a '\' which is not compatible with Microsoft's WCF stack.
```
[the remark this answers is not in the document]
Comment (free-ice) [reply ws-security-dom/src/main/java/org/apache/wss4j/dom/WSConstants.java:206]: 👍

File: ws-security-dom/src/main/java/org/apache/wss4j/dom/WSConstants.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (free-ice) [reply ws-security-dom/src/main/java/org/apache/wss4j/dom/WSConstants.java:331]: 👍

File: ws-security-dom/src/main/java/org/apache/wss4j/dom/WSConstants.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (free-ice) [reply ws-security-dom/src/main/java/org/apache/wss4j/dom/WSConstants.java]: Done

File: ws-security-dom/src/main/java/org/apache/wss4j/dom/WSConstants.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (free-ice) [reply ws-security-dom/src/main/java/org/apache/wss4j/dom/WSConstants.java]: Done

File: ws-security-dom/src/main/java/org/apache/wss4j/dom/handler/WSHandlerConstants.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (free-ice) [reply ws-security-dom/src/main/java/org/apache/wss4j/dom/handler/WSHandlerConstants.java]: Done

**Comment by free-ice** [comment]
It Looks like the comment about handling exception was removed, or at least I don't see it now ?
The execute() method throws the same exception as the X509Principal.getName() so from an Exception prespective that has not changed. If you want to to still make a change or ad a test for this let me know.

All other comments have been addressed.

**Comment by free-ice** [comment]
Also I just created a PR against spring-ws and was impressed that they have a webhook to check a CLA status and automates the process of signing it. Full process only took 1 minutes. Though I had filled one out for Apache 20 years ago when I was a committer, I downloaded a new one and the PDF is cubmersome.... I looked around to see if I could find the Pivotal CLA webhook code but was unsuccessfull.

File: ws-security-common/src/main/java/org/apache/wss4j/common/util/CommaDelimiterRfc2253Name.java [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+                    appendString = rdnString;
+                }
+                if (i == rdns.size() - 1) {
+                    commaDNBuilder.append(appendString);
+                } else {
+                    commaDNBuilder.append(", ").append(appendString);
+                }
+            }
+        } catch (InvalidNameException e) {
+            throw new IllegalArgumentException(" The distinguished name cannot be parsed : " + rfs2253String);
```
[the remark this answers is not in the document]
Comment (free-ice) [reply ws-security-common/src/main/java/org/apache/wss4j/common/util/CommaDelimiterRfc2253Name.java:69]: The execute() method throws the same exception as the X509Principal.getName() so from an possible thrown Exception prespective that has not changed.

**Pull request merged**